firestore.rules

```diff
@@ -21,18 +21,28 @@ service cloud.firestore {
       // check that requesting uid matches hostId
       return request.auth.uid == resource.data.hostId;
     }
-    function isGuest() {
+    function isGuest(eventId) {
       // check that the requesting uid is a guest
-      return request.auth.uid in resource.data.guests;
+      return exists(/databases/$(database)/documents/events/$(eventId)/guests/$(request.auth.uid));
     }
     function isSafeEventWrite() {
       // check that the hostId does not change during the write
       return request.resource.data.hostId == resource.data.hostId;
     }
-    match /events/{document=**} {
-    	allow read: if isGuest() || isHost();
-      allow create: if true;
+    match /events/{EVENTID} {
+    	allow read: if isGuest(EVENTID);
+      allow create, delete: if isHost();
       allow update: if isHost() && isSafeEventWrite();
```

Comment:
From these rules it seems like only the host can make changes to the event. Is that right?

Reply:
Yeah, I'll need to add in another rule to allow updates for availability. Im considering moving availability to another subcollection similar to how it is for event guests. So we have finer control over read / write rules for event availability.

Reply:
sounds like a good idea